src/transformers/models/gpt_sw3/tokenization_gpt_sw3.py

```diff
@@ -300,15 +327,38 @@ def encode_fast(
         return token_ids
 
     def decode_fast(self, token_ids: Union[int, List[int]]) -> str:
-        """
-        Encodes a text or batch of texts to token ids using preprocessing and the raw SP tokenizer. This has reduced
-        functionality but is often much faster.
+        """Decodes token IDs to text.
+
+        This uses preprocessing and the raw SP tokenizer. This has reduced functionality but is often much faster.
 
         Args:
-            token_ids (`int` or `List[int]`): Encoded token or text as token id(s).
+            token_ids (`int` or `List[int]`):
+                Encoded token or text as token id(s).
 
         Returns:
-            `str`: Decoded text
+            `str`:
+                Decoded text.
         """
 
         return self.sp_model.decode(token_ids)
+
+    def _build_conversation_input_ids(self, conversation: Conversation) -> List[int]:
+        """Builds the input ids for a conversation.
+
+        Args:
+            conversation (`Conversation`):
+                Conversation to build input ids for.
+
+        Returns:
+            `List[int]`:
+                Input ids for the conversation.
+        """
+        all_responses = [
+            f"User:\n{text}" if is_user else f"Bot:\n{text}" for is_user, text in conversation.iter_texts()
+        ]
+        prompt = (
+            f"{self.eos_token}{self.bos_token}\n"
+            + f"\n{self.bos_token}\n".join(all_responses)
+            + f"{self.bos_token}Bot:\n"
+        )
```

Comment:
Which format does this corresponds to? Was this the orignal format? We need to add a small comment specifying the original.[ like  this](https://github.com/ArthurZucker/transformers/blob/de889cb4b1fea75af3e4325b6c2a77e8bb3d600a/src/transformers/models/gpt_neox_japanese/tokenization_gpt_neox_japanese.py#L182)
```python 
"""This corresponds to DialoGPT variants of models."""
```

Reply:
Fair point. It's the original GPT-SW3 paper - I've included details and references in the docstring now.